### PR TITLE
*: Add priority in resource group setting

### PR DESCRIPTION
**Pull request by Connor1996**
### What problem does this PR solve?

Issue Number: ref #38825 

Problem Summary:

### What is changed and how it works?

Add a priority field in resource group setting, so we can decouple priority scheduling from RU

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Documentation

- [x] Affects user behaviors
- [x] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Add priority in resource group setting
```

### CODE REVIEW

File: parser/model/model.go [hunk 1]
```diff
@@ -1836,17 +1836,52 @@ type ResourceGroupRefInfo struct {
...
+func PriorityValueToName(value uint64) string {
+	switch value {
+	case LowPriorityValue:
+		return "LOW"
+	case MediumPriorityValue:
+		return "MEDIUM"
+	case HighPriorityValue:
+		return "HIGH"
+	default:
+		return "MEDIUM"
```
Comment (nolouch) [inline parser/model/model.go:1868]: Does it mean, expect the 1,16, others will be handled as `MEDIUM`.

Comment (Connor1996) [reply parser/model/model.go:1868]: yes

**Review by nolouch — APPROVED** [review]

**Comment by nolouch** [comment]
ci failed.

File: parser/model/model.go [hunk 2]
```diff
@@ -1836,17 +1836,52 @@ type ResourceGroupRefInfo struct {
...
+	}
+}
+
+//revive:disable:exported
+const (
+	LowPriorityValue    = 1
+	MediumPriorityValue = 8
+	HighPriorityValue   = 16
```
Comment (glorv) [inline parser/model/model.go:1876]: I think use smaller value as higher priority can let the logic in tikv slightly simpler.

File: parser/parser.y [hunk 3]
```diff
@@ -738,6 +738,10 @@ import (
 	voterConstraints      "VOTER_CONSTRAINTS"
 	voters                "VOTERS"
 	ruRate                "RU_PER_SEC"
+	priority	          "PRIORITY"
```
Comment (glorv) [inline parser/parser.y:741]: align

File: parser/parser.y [hunk 4]
```diff
@@ -1372,6 +1376,7 @@ import (
 	PlacementOptionList                    "Anomymous or direct placement option list"
 	DirectResourceGroupOption              "Subset of anonymous or direct resource group option"
 	ResourceGroupOptionList                "Anomymous or direct resource group option list"
+	ResourceGroupPriorityOption			   "Resource group priority option"
```
Comment (glorv) [inline parser/parser.y:1379]: ditto

**Review by glorv — COMMENTED** [review]
LGTM after all files are formatted.

**Review by glorv — APPROVED** [review]

**Comment by glorv** [comment]
The Dumpling test failed due to a bug in tidb-lightning, they will fix it later, here we will just skip this check.

**Pull request merged**